### ISSUE TITLE
Training loss unmatched with test performance using SmolVLA with LIBERO-Spatial

### ISSUE THREAD

**Issue by iamzzx** [huggingface/lerobot#2207]
I trained the SmolVLA with the datasets downloaded from the following link:

[https://huggingface.co/datasets/aopolin-lv/libero_spatial_no_noops_lerobot_v21](url)

As the latest version of the lerobot framework was applied, I converted the libero_spatial dataset from v21 to v30 using the scripts under the convert_dataset_v21_to_v30.py (debugged version) and augment_dataset_quantile_stats.py.

I used the SmolVLM2-500M-Video-Instruct backbone with default training configuration:

AdamWConfig(lr=0.0001, weight_decay=1e-10, grad_clip_norm=10.0, betas=(0.9, 0.95), eps=1e-08)

with the following instruction:

CUDA_VISIBLE_DEVICES=3 \
        python src/lerobot/scripts/lerobot_train.py \
        --dataset.root=../../libero_spatial_parquet \
        --policy.type=smolvla \
        --policy.repo_id=test \
        --batch_size=64 \
        --steps=200000 \
        --policy.device=cuda \
        --wandb.enable=false \
        --save_freq 10000 \
        --output_dir=outputs/test1/ \
        --job_name=libero_smolvla_ft \
        --dataset.repo_id=test

The vision encoder was freezed with the expert trained only:

The training process seems normal until the loss dropped to 0.027 after 30K steps and retained this in the next 40K steps so far. 
<img width="812" height="65" alt="Image" src="https://github.com/user-attachments/assets/0145158e-128a-477b-816c-b9146b7aea69" />

However, the success rate was always 0 across all the 10 tasks as could be see from the saved gif. 

I used the following evaluation instructions and the official smolvla_libero ckpt performed normally using this:

CUDA_VISIBLE_DEVICES=3 lerobot-eval --policy.path=outputs/test1/checkpoints/060000/pretrained_model --env.type=libero --env.task=libero_spatial --eval.batch_size=4 --eval.n_episodes=4

Anyone has a clue or any suggestions?
